torch/_dynamo/output_graph.py

```diff
@@ -814,18 +865,53 @@ def remove_unused_graphargs(self) -> None:
                 elif node.op == "call_function" and node.target is operator.getitem:
                     self.remove_node(node)
 
-        for i, node in enumerate(self.placeholders):
-            if not node.users:
-                log.debug(
-                    "REMOVE UNUSED GRAPHARG %s", node.meta["grapharg"].source.name()
-                )
-                # I'm not really sure why you need to delete these from the
-                # node since the node is going to get removed
-                if "example_value" in node.meta:
-                    del node.meta["example_value"]
-                del node.meta["grapharg"]
-                self.remove_node(node)
-                self.real_value_cache.pop(node, None)
+        def placeholder_binds_symbol(node):
+            arg = node.meta["grapharg"]
+            example = arg.example
+            if isinstance(example, torch.SymInt) and isinstance(
+                example.node.expr, sympy.Symbol
+            ):
```

Comment:
Can you explain this condition?

Reply:
We're looking for an input which is a SymInt and is a symbol and not some derived expression. I can have SymInt that expresses, e.g., s0 + s1 (or s0 // 2) and this doesn't "bind" the symbol, because you cannot infer s0/s1 from just the expression.